analyzers/tests/SonarAnalyzer.Test/Rules/RouteTemplateShouldNotStartWithSlashTest.cs

```diff
@@ -29,8 +29,156 @@ public class RouteTemplateShouldNotStartWithSlashTest
+            AspNetCoreMetadataReference.MicrosoftAspNetCoreMvcViewFeatures,
+            AspNetCoreMetadataReference.MicrosoftAspNetCoreRouting,             // For IEndpointRouteBuilder
+            AspNetCoreMetadataReference.MicrosoftAspNetCoreRazorPages,          // For RazorPagesEndpointRouteBuilderExtensions.MapFallbackToPage
+        ];
+
+    [TestMethod]
+    public void RouteTemplateShouldNotStartWithSlash_CS() =>
+        builderCS
+            .AddPaths("RouteTemplateShouldNotStartWithSlash.AspNetCore.cs")
+            .AddReferences(AspNetCoreReferences)
+            .Verify();
+
+    [DataRow("""[HttpGet("/Index1")]""", "", false)]
+    [DataRow("""[Route("/Index2")]""", "", false)]
+    [DataRow("""[HttpGet("\\Index1")]""", "", true)]
+    [DataRow("""[Route("\\Index2")]""", "", true)]
+    [DataRow("""[HttpGet("Index1/SubPath")]""", "", true)]
+    [DataRow("""[Route("Index2/SubPath")]""", "", true)]
+    [DataRow("""[Route("/IndexA")]""", """[Route("/IndexB")]""", false)]
+    [DataRow("""[Route("/IndexC")]""", """[HttpGet("/IndexD")]""", false)]
+    [DataTestMethod]
+    public void RouteTemplateShouldNotStartWithSlash_RouteAttributes(string firstAttribute, string secondAttribute, bool compliant)
+    {
+        var builder = builderCS.AddReferences(AspNetCoreReferences)
+            .AddSnippet($$"""
+                using Microsoft.AspNetCore.Mvc;
+                using Microsoft.AspNetCore.Mvc.Routing;
+
+                [Route("[controller]")]
+                public class BasicsController : Controller {{(compliant ? string.Empty : " // Noncompliant")}}
+                {
+                    {{firstAttribute}}{{(compliant ? string.Empty : " // Secondary")}}
+                    {{secondAttribute}}{{(compliant || string.IsNullOrEmpty(secondAttribute) ? string.Empty : " // Secondary")}}
+                    public ActionResult SomeAction() => View();
+                }
+                """);
+
+        if (compliant)
+        {
+            builder.VerifyNoIssueReported();
+        }
+        else
+        {
+            builder.Verify();
+        }
+    }
+
+    [DataRow("""[HttpGet("/IndexGet")]""")]
+    [DataRow("""[HttpPost("/IndexPost")]""")]
+    [DataRow("""[HttpPut("/IndexPut")]""")]
+    [DataRow("""[HttpDelete("/IndexDelete")]""")]
+    [DataRow("""[HttpPatch("/IndexPatch")]""")]
+    [DataRow("""[HttpHead("/IndexHead")]""")]
+    [DataRow("""[HttpOptions("/IndexOptions")]""")]
+    public void RouteTemplateShouldNotStartWithSlash_HttpAttributes(string attribute)
+    {
+        var builder = builderCS.AddReferences(AspNetCoreReferences)
+            .AddSnippet($$"""
+                using Microsoft.AspNetCore.Mvc;
+                using Microsoft.AspNetCore.Mvc.Routing;
+
+                public class BasicsController : Controller  // Noncompliant
+                {
+                    {{attribute}} // Secondary
+                    public ActionResult SomeAction() => View();
```

Comment:
These test cases contradict with S6934 "How to fix" option 3).

Reply:
@martin-strecker-sonarsource @antonioaversa   maybe we can exclude in S6931 routes that start with `/[controller]` as they seem pretty intentional.
WDYT?